modular_zubbers/code/modules/customization/species/proteans/organs/protean_stomach.dm

```diff
@@ -5,54 +5,49 @@
 	icon = PROTEAN_ORGAN_SPRITE
 	icon_state = "refactory"
 	organ_flags = ORGAN_ROBOTIC | ORGAN_NANOMACHINE
-	organ_traits = list(TRAIT_NOHUNGER)
-
-	/// How much max metal can we hold at any given time (In sheets). This isn't using nutrition code because nutrition code gets weird without livers.
-	var/metal_max = PROTEAN_STOMACH_FULL
-	/// How much metal are we holding currently (In sheets)
-	var/metal = PROTEAN_STOMACH_FULL
+	organ_traits = list(TRAIT_ROCK_EATER)
 	/// Multiplicative modifier to how fast we lose metal
 	var/metabolism_modifier = 1
 	COOLDOWN_DECLARE(starving_message)
 	COOLDOWN_DECLARE(damage_delay)
 
-/obj/item/organ/stomach/protean/Initialize(mapload)
-	. = ..() // Call the rest of the proc
-	metal = round(rand(PROTEAN_STOMACH_FULL/2, PROTEAN_STOMACH_FULL))
-
 /obj/item/organ/stomach/protean/on_mob_insert(mob/living/carbon/receiver, special, movement_flags)
 	. = ..()
-	RegisterSignal(receiver, COMSIG_CARBON_ATTEMPT_EAT, PROC_REF(try_stomach_eat))
 	RegisterSignal(receiver, COMSIG_MOB_AFTER_APPLY_DAMAGE, PROC_REF(damage_listener))
 
 /obj/item/organ/stomach/protean/on_mob_remove(mob/living/carbon/stomach_owner, special, movement_flags)
 	. = ..()
-	UnregisterSignal(stomach_owner, COMSIG_CARBON_ATTEMPT_EAT)
 	UnregisterSignal(stomach_owner, COMSIG_MOB_AFTER_APPLY_DAMAGE)
 
 /obj/item/organ/stomach/protean/on_life(seconds_per_tick, times_fired)
-	var/datum/species/protean/species = owner?.dna.species
-	var/obj/item/mod/control/pre_equipped/protean/suit = species.species_modsuit
-	if(owner.loc == suit)
+	if(isnull(owner.client)) // So we dont die from afk/crashing out
 		return
-	/// Zero out any nutrition. We do not use hunger in this species.
 	for(var/datum/reagent/consumable/food in reagents.reagent_list)
+		if(istype(food, /datum/reagent/consumable/nutriment/mineral))
+			continue
 		food.nutriment_factor = 0
-	. = ..()
 	handle_protean_hunger(owner, seconds_per_tick)
+	return ..()
+
+/obj/item/organ/stomach/protean/proc/health_calculations() // Heals one at 4 ticks or both damages at 2 ticks
+	var/health_amount = 4
+	if((owner.getBruteLoss() > 0) && (owner.getFireLoss() > 0))
```

Comment:
check the return values from adjustBruteLoss/adjustBurnLoss to actually see if you healed something